### PR TITLE
feat: icons v2 and neptune CSS

### PR DESCRIPTION
**Pull request by dozortsev**
## 🖼 Context

Bring migration on icons v2 and related neptune css changes in `main` branch thus in stable version.

## ✅ Checklist

- [x] Make PR title meaningful and follow [the commit lint format](https://github.com/transferwise/neptune-web/blob/master/CONTRIBUTING.md#versioning-and-commit-lint) (it will appear in the changelog)
- [x] Changes are tested and all tests pass
- [x] Changes meet [accessibility standards](https://github.com/transferwise/neptune-web/blob/main/ACCESSIBILITY.md) and there are no violations in the console
- [x] Changes work in all supported browsers (don't forget IE11)
- [x] You've updated the documentation if necessary

### CODE REVIEW

File: packages/components/babel.config.js [hunk 1]
```diff
@@ -79,6 +79,7 @@ module.exports = {
     '@babel/plugin-proposal-class-properties',
     '@babel/plugin-transform-react-jsx',
     '@babel/plugin-transform-runtime',
+    'transform-react-remove-prop-types',
```
Comment (andreapiras) [inline packages/components/babel.config.js:82]: This cause issues in the past with props checking being removed in production

Comment (andreapiras) [reply packages/components/babel.config.js:82]: How is that related with icons?

File: packages/components/src/checkbox/Checkbox.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 import Types from 'prop-types';
-import CheckboxButton from '../common/CheckboxButton';
+import CheckboxButton from '../checkboxButton';
```
Comment (andreapiras) [inline packages/components/src/checkbox/Checkbox.js:3]: I would exclude all non icon related changes

File: packages/components/src/common/CloseButton/CloseButton.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  </button>
+);
+
+CloseButton.propTypes = {
+  onClick: Types.func.isRequired,
+  className: Types.string,
+};
+
+CloseButton.defaultProps = {
+  className: null,
```
Comment (andreapiras) [inline packages/components/src/common/CloseButton/CloseButton.js:24]: since the type expected is string should this be '' rather than null?

File: packages/components/src/common/CloseButton/CloseButton.less [hunk 4]
```diff
@@ -0,0 +1,19 @@
+@import 'bootstrap/less/variables';
```
Comment (andreapiras) [inline packages/components/src/common/CloseButton/CloseButton.less:1]: Should we get rid of bootstrap references?

File: packages/components/src/drawer/Drawer.js [hunk 5]
```diff
@@ -31,12 +31,12 @@ const Drawer = ({ open, position, onClose, children, headerTitle, footerContent
           >
             <div
               role="button"
-              className="drawer-header--close"
+              className="drawer-header close"
```
Comment (andreapiras) [inline packages/components/src/drawer/Drawer.js:34]: has `drawer-header--close` been removed from less?

File: packages/components/src/flowNavigation/FlowNavigation.less [hunk 6]
```diff
@@ -77,27 +72,20 @@
...
   }
 
-  &__close-button {
-    display: inline;
-    transition: color 0.15s ease-in-out;
-    color: var(--color-text-link);
-    outline: 0;
-    padding: 0;
+  &__right-content {
+    display: flex;
```
Comment (andreapiras) [inline packages/components/src/flowNavigation/FlowNavigation.less:76]: Can we not reuse the flex classes to achieve this? It would be more testable and produce less untestable css. example `d-flex `

File: packages/components/src/flowNavigation/FlowNavigation.less [hunk 7]
```diff
@@ -77,27 +72,20 @@
...
+    align-items: center;
+    justify-content: flex-end;
+  }
 
-    &:hover {
-      color: var(--color-text-link-hover);
-    }
+  &__right-content > .avatar-container {
+    display: flex;
```
Comment (andreapiras) [inline packages/components/src/flowNavigation/FlowNavigation.less:82]: maybe we cann pass d-flex

File: packages/components/src/flowNavigation/FlowNavigation.less [hunk 8]
```diff
@@ -77,27 +72,20 @@
...
-    &::after {
-      cursor: default;
-      content: '';
-      height: 100%;
-      border-left: solid 1px var(--color-border-neutral);
-      left: -48px;
-      position: relative;
-    }
+  &__right-content > .avatar-container > .circle {
+    display: flex;
```
Comment (andreapiras) [inline packages/components/src/flowNavigation/FlowNavigation.less:86]: as above

File: packages/components/src/flowNavigation/avatarWrapper/AvatarWrapper.js [hunk 9]
```diff
@@ -2,7 +2,7 @@ import React, { useState, useEffect } from 'react';
 import Types from 'prop-types';
 import Avatar from '../../avatar';
 
-import './avatarWrapper.css';
```
Comment (andreapiras) [inline packages/components/src/flowNavigation/avatarWrapper/AvatarWrapper.js:5]: has avatarWrapper.less been removed?

File: packages/components/src/image/Image.less [hunk 10]
```diff
@@ -3,10 +3,12 @@
   width: auto;
   // This prevents content to shrink when shrink=false and stretch=true due to width: 100%
   min-width: min-content;
+
```
Comment (andreapiras) [inline packages/components/src/image/Image.less:6]: For future PR let's try to keep the changes as minimun to reduce CR fatigue

File: packages/components/src/pagination/Pagination.less [hunk 11]
```diff
@@ -0,0 +1,50 @@
+.tw-pagination {
```
Comment (andreapiras) [inline packages/components/src/pagination/Pagination.less:1]: Are these files not on main already?

File: packages/components/src/popover/Popover.less [hunk 12]
```diff
@@ -1,9 +1,11 @@
-html.ios-click {
```
Comment (andreapiras) [inline packages/components/src/popover/Popover.less:1]: ?

File: packages/components/src/select/Select.js [hunk 13]
```diff
@@ -13,8 +15,6 @@ import {
 } from '../common/domHelpers';
 import { addClassAndTriggerReflow, removeClass } from './domHelpers';
 
-import './Select.css';
```
Comment (andreapiras) [inline packages/components/src/select/Select.js:16]: Is this intentional? Looks like Select.less hasn't been removed

File: packages/components/src/upload/steps/completeStep/completeStep.spec.js [hunk 14]
```diff
@@ -1,8 +1,7 @@
 import React from 'react';
 import { shallow } from 'enzyme';
 import CompleteStep from './';
-import AlertCircleIcon from '@transferwise/icons/react/alert-circle';
-import DocumentIcon from '@transferwise/icons/react/document';
+import { AlertCircle as AlertCircleIcon, Document as DocumentIcon } from '@transferwise/icons';
```
Comment (andreapiras) [inline packages/components/src/upload/steps/completeStep/completeStep.spec.js:4]: Why are we changing the names?

File: packages/css/src/less/select.less [hunk 15]
```diff
@@ -1,17 +1,16 @@
 @import (reference) '@transferwise/neptune-css/src/variables/neptune-tokens.less';
```
Comment (andreapiras) [inline packages/css/src/less/select.less:1]: Should this live next to the component?

File: packages/components/src/alert/Alert.js [hunk 16]
```diff
@@ -4,6 +4,7 @@ import Types from 'prop-types';
 import classNames from 'classnames';
 import requiredIf from 'react-required-if';
 import { Size, MessageType, ArrowPosition } from '../common';
+import { Cross as CrossIcon } from '@transferwise/icons';
```
Comment (andreapiras) [inline packages/components/src/alert/Alert.js:7]: LEt's chat about this renaming

File: packages/components/src/chevron/Chevron.spec.js [hunk 17]
```diff
@@ -1,39 +1,48 @@
...
-  let wrapper;
+describe('Chevron', () => {
+  describe('test rotation', () => {
+    it('top', () => {
+      const { container } = render(<Chevron orientation={Orientation.TOP} />);
 
-  beforeEach(() => {
-    wrapper = shallow(<Chevron {...props} />);
-  });
+      expect(container.querySelector('.tw-icon-chevron-up.top')).not.toBe(null);
```
Comment (andreapiras) [inline packages/components/src/chevron/Chevron.spec.js:13]: Let's rely less on classes and use more built in func

File: packages/components/src/common/CloseButton/CloseButton.js [hunk 18]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import Types from 'prop-types';
+import classNames from 'classnames';
+import { Cross as CrossIcon } from '@transferwise/icons';
+
+import './CloseButton.css';
+
+export const CloseButton = ({ onClick, className }) => (
+  <button
```
Comment (andreapiras) [inline packages/components/src/common/CloseButton/CloseButton.js:9]: This can be replaced with Button type link

File: packages/components/src/common/Option/Option.js [hunk 19]
```diff
@@ -3,8 +3,6 @@ import Types from 'prop-types';
 import classNames from 'classnames';
 import requiredIf from 'react-required-if';
 
-import './Option.css';
```
Comment (andreapiras) [inline packages/components/src/common/Option/Option.js:6]: Has this been moved somewhere elase

File: packages/components/src/flowNavigation/FlowNavigation.js [hunk 20]
```diff
@@ -62,7 +52,7 @@ const FlowNavigation = ({
         </div>
       }
       rightContent={
-        <div className="tw-flow-navigation__avatar-with-close-button m-lg-t-1">
+        <div className="tw-flow-navigation__right-content m-lg-t-1">
```
Comment (andreapiras) [inline packages/components/src/flowNavigation/FlowNavigation.js:55]: you can use .d-flex justifyContent etc

Comment (andreapiras) [reply packages/components/src/flowNavigation/FlowNavigation.js:55]: _flex.less

File: packages/components/src/moneyInput/MoneyInput.less [hunk 21]
```diff
@@ -27,3 +27,22 @@
     position: relative;
   }
 }
+
+// Fix chevron icon alignment (icons v2) in currency select button
+.tw-money-input .amount-currency-select-btn > .tw-select > .btn {
+  display: flex;
```
Comment (andreapiras) [inline packages/components/src/moneyInput/MoneyInput.less:33]: use utilities

File: packages/components/src/pagination/Pagination.js [hunk 22]
```diff
@@ -0,0 +1,146 @@
+import React from 'react';
```
Comment (andreapiras) [inline packages/components/src/pagination/Pagination.js:1]: Should not be part of this PR

File: packages/components/src/select/option/Option.js [hunk 23]
```diff
@@ -4,19 +4,24 @@ import Types from 'prop-types';
...
     classes.map((className) => classNames[className] || className).join(' ');
-  const iconClass = currency
+  const currencyClassNames = currency
     ? `${style([
         'currency-flag',
         `currency-flag-${currency.toLowerCase()}`,
         `${selected ? 'hidden-xs' : ''}`,
       ])}`
-    : `${style(['icon'])} ${icon}`;
+    : undefined;
```
Comment (andreapiras) [inline packages/components/src/select/option/Option.js:13]: null?

File: packages/components/src/switch/Switch.less [hunk 24]
```diff
@@ -8,26 +8,33 @@
   padding: 2px;
```
Comment (andreapiras) [inline packages/components/src/switch/Switch.less:8]: Remove from PR

File: packages/css/package.json [hunk 25]
```diff
@@ -55,6 +55,11 @@
     "prop-types": "^15.7.2",
```
Comment (andreapiras) [inline packages/css/package.json:55]: Remove from Pr

File: packages/css/pages/app.css [hunk 26]
```diff
@@ -1,10 +1,14 @@
+body {
```
Comment (andreapiras) [inline packages/css/pages/app.css:1]: Remove from PR

File: packages/css/src/less/media.less [hunk 27]
```diff
@@ -2,10 +2,6 @@
   display: flex;
 }
 
-.media + .media {
```
Comment (andreapiras) [inline packages/css/src/less/media.less:5]: This is a massive sitewide change

File: packages/css/src/less/tick.less [hunk 28]
```diff
@@ -0,0 +1,32 @@
...
+  height: 8px;
+
+  &::before,
+  &::after {
+    content: '';
+    position: absolute;
+    background: var(--color-background-screen);
+    transform: translateX(0.5px) rotate(-45deg);
+    transform-origin: left bottom;
+    left: 8px;
```
Comment (andreapiras) [inline packages/css/src/less/tick.less:14]: Should we use spacing vars here?

File: packages/docs/liveEditorCode/pagination.code.js [hunk 29]
```diff
@@ -0,0 +1,13 @@
+() => {
```
Comment (andreapiras) [inline packages/docs/liveEditorCode/pagination.code.js:1]: shoudl not be in Pr

File: packages/docs/pages/components/Image.mdx [hunk 30]
```diff
@@ -0,0 +1,75 @@
+import { LiveEditorBlock, GeneratePropsTable } from '../../utils';
```
Comment (andreapiras) [inline packages/docs/pages/components/Image.mdx:1]: Should not be in PR

File: packages/docs/pages/components/inputs/DateInput.mdx [hunk 31]
```diff
@@ -6,5 +6,5 @@ import code from '../../../liveEditorCode/dateInput.code';
 <GeneratePropsTable componentName="DateInput" />
 
 export const meta = {
-  name: 'Date Input',
+  name: 'DateInput',
```
Comment (andreapiras) [inline packages/docs/pages/components/inputs/DateInput.mdx:9]: Should not be in PR

File: packages/docs/pages/components/inputs/DateLookup.mdx [hunk 32]
```diff
@@ -6,5 +6,5 @@ import code from '../../../liveEditorCode/dateLookup.code';
 <GeneratePropsTable componentName="DateLookup" />
 
 export const meta = {
-  name: 'Date Lookup',
+  name: 'DateLookup',
```
Comment (andreapiras) [inline packages/docs/pages/components/inputs/DateLookup.mdx:9]: Should not be in PR

File: packages/docs/pages/components/navigation/Pagination.mdx [hunk 33]
```diff
@@ -0,0 +1,10 @@
+import { LiveEditorBlock, GeneratePropsTable } from '../../../utils';
+import { Pagination } from '@transferwise/components';
```
Comment (andreapiras) [inline packages/docs/pages/components/navigation/Pagination.mdx:2]: Should not be in PR

File: packages/dynamic-flows/babel.config.js [hunk 34]
```diff
@@ -53,6 +53,7 @@ module.exports = {
     '@babel/plugin-proposal-class-properties',
     '@babel/plugin-transform-react-jsx',
     '@babel/plugin-transform-runtime',
+    'transform-react-remove-prop-types',
```
Comment (andreapiras) [inline packages/dynamic-flows/babel.config.js:56]: Should not be in PR

File: packages/dynamic-flows/package.json [hunk 35]
```diff
@@ -53,6 +53,7 @@
     "@transferwise/neptune-tokens": "^1.0.0",
     "@transferwise/test-config": "^2.0.0",
     "babel-loader": "^8.1.0",
+    "babel-plugin-transform-react-remove-prop-types": "^0.4.24",
```
Comment (andreapiras) [inline packages/dynamic-flows/package.json:56]: Should not be in PR

File: packages/test-config/package.json [hunk 36]
```diff
@@ -27,6 +27,7 @@
     "@testing-library/user-event": "^10.2.3",
     "enzyme": "^3.11.0",
     "enzyme-adapter-react-16": "^1.15.2",
+    "jest-runner-eslint": "^0.9.0",
```
Comment (andreapiras) [inline packages/test-config/package.json:30]: Should not be in PR

File: packages/components/src/flowNavigation/FlowNavigation.js (hunk 20 above)
Comment (dozortsev) [reply packages/components/src/flowNavigation/FlowNavigation.js:55]: this is really a bit out of scope of this PR, this was done in the original migration PR,

we can fix that later in one go in another day along with other similar places

File: packages/components/src/common/CloseButton/CloseButton.js (hunk 3 above)
Comment (dozortsev) [reply packages/components/src/common/CloseButton/CloseButton.js:24]: string value can be `null` value as well, in other components `className` props have `null` as default as well

File: packages/components/src/image/Image.less (hunk 10 above)
Comment (dozortsev) [reply packages/components/src/image/Image.less:6]: well, yeah, make sense,

but this done not by me in `beta` branch 

https://github.com/transferwise/neptune-web/commit/e99ee90991a97f990db505cbdbc2b409492c589a

Comment (dozortsev) [reply packages/components/src/image/Image.less:6]: I think this is something that we want have in `main`

File: packages/components/src/moneyInput/MoneyInput.less (hunk 21 above)
Comment (dozortsev) [reply packages/components/src/moneyInput/MoneyInput.less:33]: yeah, I agree that probably it make sense to use utils here

but this has been done not by me in `beta` branch https://github.com/transferwise/neptune-web/pull/479/files

File: packages/components/src/pagination/Pagination.js (hunk 22 above)
Comment (dozortsev) [reply packages/components/src/pagination/Pagination.js:1]: well, yeah, but Pagination component was added into `beta` branch https://github.com/transferwise/neptune-web/pull/506 so 🤷‍♂️

File: packages/components/src/pagination/Pagination.less (hunk 11 above)
Comment (dozortsev) [reply packages/components/src/pagination/Pagination.less:1]: no, only in `beta` branch, see previous https://github.com/transferwise/neptune-web/pull/563/files#r487129079

File: packages/components/src/select/Select.js (hunk 13 above)
Comment (dozortsev) [reply packages/components/src/select/Select.js:16]: it's moved in neptune-css package

File: packages/components/src/common/CloseButton/CloseButton.less [hunk 37]
```diff
@@ -0,0 +1,19 @@
...
+
+.tw-close-button {
+  transition: color 0.15s ease-in-out;
+  color: @brand-light-blue;
+  outline: 0;
+  padding: 0 0 0 20px;
+  margin-left: 20px;
+  border-left: 1px solid @brand-smoke-plus-10;
```
Comment (anttio) [inline packages/components/src/common/CloseButton/CloseButton.less:9]:
```suggestion
  border-left: 1px solid var(--color-border-neutral);
```

File: packages/components/src/common/CloseButton/CloseButton.less [hunk 38]
```diff
@@ -0,0 +1,19 @@
+@import 'bootstrap/less/variables';
+
+.tw-close-button {
+  transition: color 0.15s ease-in-out;
+  color: @brand-light-blue;
```
Comment (anttio) [inline packages/components/src/common/CloseButton/CloseButton.less:5]:
```suggestion
  color: var(--color-control-accent);
```

File: packages/components/src/common/CloseButton/CloseButton.less [hunk 39]
```diff
@@ -0,0 +1,19 @@
...
+  outline: 0;
+  padding: 0 0 0 20px;
+  margin-left: 20px;
+  border-left: 1px solid @brand-smoke-plus-10;
+
+  &:hover {
+    color: @brand-light-blue-minus-10;
```
Comment (anttio) [inline packages/components/src/common/CloseButton/CloseButton.less:12]:
```suggestion
    color: var(--color-control-accent-hover);
```

File: packages/components/src/switch/Switch.less (hunk 24 above)
Comment (dozortsev) [reply packages/components/src/switch/Switch.less:8]: this was merged straight into `beta` branch and still a necessary change to be in `main`

https://github.com/transferwise/neptune-web/commit/e99ee90991a97f990db505cbdbc2b409492c589a#diff-036826d5056bba751c35e2bdc526068a

File: packages/dynamic-flows/babel.config.js (hunk 34 above)
Comment (dozortsev) [reply packages/dynamic-flows/babel.config.js:56]: yep, fix in https://github.com/transferwise/neptune-web/pull/578

File: packages/css/src/less/core/_scaffolding.less [hunk 40]
```diff
@@ -94,6 +94,10 @@ a,
   }
 }
 
+html.ios-click {
+  cursor: pointer;
```
Comment (dozortsev) [inline packages/css/src/less/core/_scaffolding.less:98]: seems like has nothing todo with icons but it was added directly into `beta` branch https://github.com/transferwise/neptune-web/pull/505

File: packages/dynamic-flows/package.json (hunk 35 above)
Comment (dozortsev) [reply packages/dynamic-flows/package.json:56]: yep, fix in https://github.com/transferwise/neptune-web/pull/578

File: packages/test-config/package.json (hunk 36 above)
Comment (dozortsev) [reply packages/test-config/package.json:30]: yep, fix in https://github.com/transferwise/neptune-web/pull/578

File: packages/components/src/ssr.spec.js [hunk 41]
```diff
@@ -40,6 +40,7 @@ describe('Server side rendering', () => {
     body: 'body',
     onClose: jest.fn(),
     onRemove: jest.fn(),
+    checked: false,
```
Comment (dozortsev) [inline packages/components/src/ssr.spec.js:43]: was added along with switch component https://github.com/transferwise/neptune-web/pull/401

Comment (dozortsev) [reply packages/components/src/ssr.spec.js:43]: you didn't add this when you ported switch into `main` branch so do we need this?

File: packages/components/src/badge/Badge.less [hunk 42]
```diff
@@ -33,13 +33,12 @@
...
-        transparent 0,
-        transparent calc(var(--badge-size) / 2 + var(--badge-mask)),
-        black calc(var(--badge-size) / 2 + var(--badge-mask) + 0.5px)
-      );
+    mask-image: radial-gradient(
+      circle at top calc(100% - var(--badge-mask-offset)) left calc(100% - var(--badge-mask-offset)),
+      transparent 0,
+      transparent calc(var(--badge-size) / 2 + var(--badge-mask)),
+      black calc(var(--badge-size) / 2 + var(--badge-mask) + 0.5px)
+    );
```
Comment (dozortsev) [inline packages/components/src/badge/Badge.less:41]: for some reason this code formatting change just gets added here

**Comment by helendurranttw** [comment]
Superseded.
